resources/aws_ec2.go

```diff
@@ -971,25 +995,38 @@ func (obj *AwsEc2Res) snsListener(listenAddr string) (net.Listener, error) {
 }
 
 // snsPostHandler listens for posts on the SNS Endpoint.
-// TODO: download pem and check message against signature
 func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 	if req.Method != "POST" {
 		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
 		return
 	}
-	// decode json
+	// Decode the post. If an error is produced we either ignore the post,
+	// or if ErrorJSONStrict is true, send the error through awsChan so
+	// Watch() can return the error and the engine can shut down.
 	decoder := json.NewDecoder(req.Body)
 	var post postData
 	if err := decoder.Decode(&post); err != nil {
-		http.Error(w, "Bad request", http.StatusBadRequest)
-		select {
-		case obj.awsChan <- &chanStruct{
-			err: errwrap.Wrapf(err, "error decoding incoming POST, check struct formatting"),
-		}:
-		case <-obj.closeChan:
+		if obj.ErrorJSONStrict {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "error decoding incoming POST, check struct formatting"),
+			}:
+			case <-obj.closeChan:
+			}
+			return
 		}
+		log.Printf("%s: error decoding post: %s", obj, err)
```

Comment:
Don't these two need to be moved up to the top? If ErrorJSONStrict we don't get the log message!

Reply:
I figured we wanted one or the other, but this makes sense. Changed.